### PR TITLE
Fixes to HAL and TM Packer and CDH

### PR DESCRIPTION
**Pull request by Akshat412**
Some bugs that have been caught over the last few days in testing with the GS:

- Changes to TM packing for HBs and TM STORAGE
- Comms now sets internal file ID definition based on filename
- Precondition for UPLINK TIME REFERENCE updated to use existing CPy packages, and have more mission-specific scope
- DH.get_storage_info has more error checking
- CMD Processor test for time preconditions updated to match the new precondition check

### CODE REVIEW

File: tests/test_cmd_processor.py [hunk 1]
```diff
@@ -221,7 +221,7 @@ def test_valid_time_format():
     assert valid_time_format(1589387500)
 
     # Test edge cases and failing
-    assert valid_time_format(0)
+    assert not valid_time_format(0)  # outside of valid mission range
```
Comment (Akshat412) [inline tests/test_cmd_processor.py:224]: This is not supported by struct_time in CircuitPython.

File: flight/apps/command/preconditions.py [hunk 2]
```diff
@@ -23,14 +21,23 @@ def valid_time_format(*args) -> bool:
...
+    limit the time reference to at least this date.
+
+    Can also update this to be our launch date,
+    although that will lead to issues in testing
+    while in development.
+    """
+
+    # Check that the timestamp is above Jan 1st 2020
+    # Check that the timestamp is below Jan 1st 2080
+    if 1577854800 < time_reference < 3471310800:
```
Comment (perrintong325) [inline flight/apps/command/preconditions.py:38]: can we double check these values? I'm not confident that I fully know how epoch time actually works in CPY, but these values are only 12AM in EST, they are 5AM in GMT which I'm assuming that's how it is interpreted by default?

File: flight/apps/telemetry/packing.py [hunk 3]
```diff
@@ -81,10 +82,10 @@ def pack_tm_heartbeat(cls):
         if not cls._TM_AVAILABLE:
             cls._TM_AVAILABLE = True
 
-        cls._FRAME = bytearray(227 + 4)  # pre-allocated buffer for packing
+        cls._FRAME = bytearray(229 + 4)  # pre-allocated buffer for packing
```
Comment (perrintong325) [inline flight/apps/telemetry/packing.py:85]: magic number

File: flight/apps/telemetry/packing.py [hunk 4]
```diff
@@ -81,10 +82,10 @@ def pack_tm_heartbeat(cls):
...
         cls._FRAME[0] = const(0x01) & 0xFF  # message ID
         cls._FRAME[1:3] = pack_unsigned_short_int([const(0x00)], 0)  # sequence count
-        cls._FRAME[3] = const(227) & 0xFF  # packet length
+        cls._FRAME[3] = const(229) & 0xFF  # packet length
```
Comment (perrintong325) [inline flight/apps/telemetry/packing.py:88]: magic number

File: flight/apps/telemetry/packing.py [hunk 5]
```diff
@@ -346,10 +347,10 @@ def pack_tm_hal(cls):
             cls._TM_AVAILABLE = True
 
         # TODO: Frame definition for TM_HAL
-        cls._FRAME = bytearray(13 + 4)  # pre-allocated buffer for packing
+        cls._FRAME = bytearray(14 + 4)  # pre-allocated buffer for packing
```
Comment (perrintong325) [inline flight/apps/telemetry/packing.py:350]: magic number

File: flight/apps/telemetry/packing.py [hunk 6]
```diff
@@ -346,10 +347,10 @@ def pack_tm_hal(cls):
...
         cls._FRAME[0] = const(0x02) & 0xFF  # message ID
```
Comment (perrintong325) [inline flight/apps/telemetry/packing.py:351]: magic number

**Review by perrintong325 — CHANGES_REQUESTED** [review]

File: flight/apps/command/preconditions.py (hunk 2 above)
Comment (Akshat412) [reply flight/apps/command/preconditions.py:38]: Good catch. We're sticking with UTC time for the mission, so I'll update these.

**Comment by Akshat412** [comment]
This is a good opportunity to start removing magic numbers where possible lol. Good catch on those, I will add constants for all message lengths.

**Comment by Akshat412** [comment]
Fixed all magic numbers.

**Review by s026134 — CHANGES_REQUESTED** [review]

File: flight/apps/command/preconditions.py [hunk 7]
```diff
@@ -23,14 +21,23 @@ def valid_time_format(*args) -> bool:
...
+    limit the time reference to at least this date.
+
+    Can also update this to be our launch date,
+    although that will lead to issues in testing
+    while in development.
+    """
+
+    # Check that the timestamp is above Jan 1st 2020 UTC
+    # Check that the timestamp is below Jan 1st 2030 UTC
+    if 1577836800 < time_reference < 1893456000:
```
[the remark this answers is not in the document]
Comment (Akshat412) [reply flight/apps/command/preconditions.py:38]: Yeah fair enough

**Review by s026134 — APPROVED** [review]

**Review by perrintong325 — APPROVED** [review]
